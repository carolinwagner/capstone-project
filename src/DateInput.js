import React from 'react'
import styled from 'styled-components/macro'

const DateInput = ({ question, register }) => {
  const minToday = question?.validationNative?.minToday
  const getToday = () => new Date().toISOString().split('T')[0]
  return (
    <StyledDateInput
      id={question.name}
      type="date"
      name={question?.name || 'defaultDateInput'}
      ref={register(question.validationHookForm)}
      {...(minToday && { min: getToday() })}
    />
  )
}

export default DateInput

const StyledDateInput = styled.input`
  font-family: 'lato';
  width: 100%;
  padding: 12px 20px;
  margin: 20px 0;
  display: inline-block;
  border: 1px solid var(--lightblue);
  border-radius: 4px;
`
