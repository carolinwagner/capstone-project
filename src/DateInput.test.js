import React from 'react'
import '@testing-library/jest-dom/extend-expect'
import { render } from '@testing-library/react'
import DateInput from './DateInput'

describe('DateInput', () => {
  it('renders correctly', () => {
    const container = render(<DateInput />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
