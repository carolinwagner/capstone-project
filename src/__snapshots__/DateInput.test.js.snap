// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateInput renders correctly 1`] = `undefined`;
